stats/cloud/client.go

```diff
@@ -201,3 +214,26 @@ func shouldRetry(resp *http.Response, err error, attempt, maxAttempts int) bool
 
 	return false
 }
+
+func shouldAddIdempotencyKey(req *http.Request) bool {
+	switch req.Method {
+	case http.MethodGet, http.MethodHead, http.MethodOptions, http.MethodTrace:
+		return false
+	default:
+		return req.Header.Get(k6IdempotencyKeyHeader) == ""
```

Comment:
this should be just `true` at this point

Reply:
I will leave it as-is for now, as we can use the function in other place, not necessary only inside NewRequest.

Reply:
@MStoykov It's not "for the future", it's about the functional logic. The function mean *should* we add idempotency key? The answer is no for GET, HEAD, OPTIONS, TRACE requests, or request which *has* an idempotency key already. Otherwise, yes.

The fact that we currently use it in NewRequest, which make the request can't not have an idempotency key already, mustn't affect the logic of function.